### PR TITLE
Document the varchar data type.

### PR DESCRIPTION
**Pull request by kovrus**
## Summary of the changes / Why this improves CrateDB

The docs varchar PR will be merged last into the `varchar` feature branch. 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/general/ddl/data-types.rst [hunk 1]
```diff
@@ -82,28 +82,110 @@ A basic boolean type. Accepting ``true`` and ``false`` as values. Example::
...
+
+    cr> CREATE TABLE users (id varchar(2), name varchar));
+    CREATE OK, 1 row affected (... sec)
+
+::
+
+    cr> INSERT INTO users (id, name) VALUES ('1361', 'john doe')
+    SQLActionException[value too long for type character varying(2)]
+
+Unless the excess characters are all spaces, in which case the string will
```
Comment (kovrus) [inline docs/general/ddl/data-types.rst:122]: pg documentation says the following about this behaviour: `. (This somewhat bizarre exception is required by the SQL standard.)`

File: docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -82,28 +82,116 @@ A basic boolean type. Accepting ``true`` and ``false`` as values. Example::
...
+    cr> CREATE TABLE users (id varchar, name varchar(6));
+    CREATE OK, 1 row affected (... sec)
+
+::
+
+    cr> INSERT INTO users (id, name) VALUES ('1361', 'john doe')
+    SQLActionException[SQLParseException: 'john doe' is too long for the text type of length: 6]
+
+Unless the excess characters are all spaces, in this case the string literal
+will be truncated to the specified length.
```
Comment (infoverload) [inline docs/general/ddl/data-types.rst:123]: If the excess characters are all spaces, the string literal will be truncated 
to the specified length.

File: docs/general/ddl/data-types.rst [hunk 3]
```diff
@@ -82,28 +82,116 @@ A basic boolean type. Accepting ``true`` and ``false`` as values. Example::
     cr> drop table my_bool_table;
     DROP OK, 1 row affected (... sec)
 
+.. _character-data-types:
+
+Character Types
+===============
+
+The general-purpose character data types available in CrateDB.
```
Comment (infoverload) [inline docs/general/ddl/data-types.rst:90]: These are general purpose character data types available in CrateDB.

File: docs/general/ddl/data-types.rst [hunk 4]
```diff
@@ -82,28 +82,116 @@ A basic boolean type. Accepting ``true`` and ``false`` as values. Example::
...
+
+Columns of below listed character data types can also be analyzed. By default
```
Comment (infoverload) [inline docs/general/ddl/data-types.rst:92]: I think we can remove "Columns of below listed character data types can also be analyzed."

Comment (kovrus) [reply docs/general/ddl/data-types.rst]: it was not correct actually, changed it now and pushed the fixup.

**Pull request merged**